lib/formulator.ex

```diff
@@ -83,16 +116,60 @@ defmodule Formulator do
   defp build_input(form, field, options, error) do
     input_type = options[:as] || :text
     input_class = options[:class] || ""
+
     options =
       options
+      |> add_validation_attributes(form, field)
+      |> add_format_validation_attribute(form, field)
       |> Keyword.delete(:as)
       |> Keyword.put(:class, add_error_class(input_class, error.class))
 
     apply(Phoenix.HTML.Form, input_function(input_type), [form, field, options])
   end
 
+  defp add_validation_attributes(options, %{impl: impl, source: %{}} = form, field) when is_atom(impl) do
+    if should_validate?(options) do
+      form
+      |> Phoenix.HTML.Form.input_validations(field)
+      |> Keyword.merge(options)
+      |> Keyword.delete(:validate)
+    else
+      options
+    end
+  end
+  defp add_validation_attributes(options, _, _), do: options
+
+  defp add_format_validation_attribute(options, %{impl: impl, source: %{}} = form, field) when is_atom(impl) do
+    with true <- should_validate_regex?(options),
+      {:format, regex} <- form.source.validations[field]
+    do
+      options
+      |> Keyword.put_new(:pattern, Regex.source(regex))
+      |> Keyword.delete(:validate_regex)
+    else
+      _ -> options
+    end
+  end
+  defp add_format_validation_attribute(options, _, _), do: options
+
+  defp should_validate?(options) do
+    Enum.any?([
+      options[:validate] == true,
+      Application.get_env(:formulator, :validate, true) == true,
+    ])
+  end
+
+  defp should_validate_regex?(options) do
+    Enum.any?([
```

Comment:
This logic is duplicated with the only difference being the key that is used. What do you think about extracting this into something like `option_enabled?(:validate_regex)`

Reply:
👍